### PR TITLE
Add prettier and husky hook for automatic formatting

### PR DESCRIPTION
**Pull request by echanlee**
Must run `yarn install` to enable automatic formatting. Should run prettier linter when performing `git commit`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,5 +94,8 @@
         "compile": "tsc -p ./",
         "update-vscode": "node ./node_modules/vscode/bin/install",
         "postinstall": "node ./node_modules/vscode/bin/install"
+    },
+    "lint-staged": {
+        "**/*": "prettier --write --ignore-unknown"
```
Comment (zcai1) [inline package.json:99]: How about `prettier --write --ignore-unknown src/*.js` instead of having a large `.prettierignore` file?

**Review by wmdietl — APPROVED** [review]
Thanks!

**Review by wmdietl — COMMENTED** [review]
One question: what does a developer need to do after editing code?
Can you add a section to the README about how to check formatting and reformat code?

**Comment by wmdietl** [comment]
> One question: what does a developer need to do after editing code?
> Can you add a section to the README about how to check formatting and reformat code?

I merged #84, so you can mention anything in the developer's section.

**Comment by echanlee** [comment]
> One question: what does a developer need to do after editing code?
> Can you add a section to the README about how to check formatting and reformat code?

The developer doesn't need to do anything differently. When performing a `git commit` the linter will run and automatically reformat all staged files. Do I still need to update the README file to explain this?

**Review by wmdietl — APPROVED** [review]

**Comment by wmdietl** [comment]
Thanks! I'm fine with not mentioning for now. Let's see if anyone gets surprised by that behavior :-)

**Pull request merged**